include/proxsuite/proxqp/dense/preconditioner/ruiz.hpp

```diff
@@ -86,32 +86,44 @@ ruiz_scale_qp_in_place( //
       for (isize k = 0; k < n; ++k) {
         switch (sym) {
           case Symmetry::upper: { // upper triangular part
-            delta(k) = T(1) / (sqrt(std::max({
-                                 infty_norm(H.col(k).head(k)),
-                                 infty_norm(H.row(k).tail(n - k)),
-                                 infty_norm(A.col(k)),
-                                 infty_norm(C.col(k)),
-                               })) +
-                               machine_eps);
+            T aux = sqrt(std::max({
+              infty_norm(H.col(k).head(k)),
+              infty_norm(H.row(k).tail(n - k)),
+              infty_norm(A.col(k)),
+              infty_norm(C.col(k)),
+            }));
+            if (aux == T(0)) {
+              aux = T(1);
+            }
+
+            delta(k) = T(1) / (aux + machine_eps);
             break;
           }
           case Symmetry::lower: { // lower triangular part
-            delta(k) = T(1) / (sqrt(std::max({
-                                 infty_norm(H.row(k).head(k)),
-                                 infty_norm(H.col(k).tail(n - k)),
-                                 infty_norm(A.col(k)),
-                                 infty_norm(C.col(k)),
-                               })) +
-                               machine_eps);
+
+            T aux = sqrt(std::max({
+              infty_norm(H.col(k).head(k)),
+              infty_norm(H.col(k).tail(n - k)),
+              infty_norm(A.col(k)),
+              infty_norm(C.col(k)),
+            }));
+            if (aux == T(0)) {
+              aux = T(1);
+            }
+            delta(k) = T(1) / (aux + machine_eps);
```

Comment:
@Bambade I don't understand the switch from aux = 0 to aux = 1. Could comment on it?

Reply:
idem (cf. above)

Reply:
I would rather put if(aux == 0) delta(k) = 1 else delta = normal value
```suggestion
            if (aux == T(0)) {
              delta(k) = T(1);
            }
            else {
              delta(k) = T(1) / (aux + machine_eps);
            }
```

Reply:
@Bambade what do you think?

Reply:
Yes, it's better indeed!

Reply:
I will let you apply the change where it is needed.

Reply:
OK!